extending-valtimo/dashboard/custom-data-source.md

```diff
@@ -0,0 +1,53 @@
+# Creating custom data sources
+
+## Back-end
+
+The dependency `com.ritense.valtimo:dashboard` is required to develop a data source in the Valtimo back-end.
+
+### Data source specification
+
+First, a data source specification using the `@WidgetDataSource` annotation needs to be created. Below is an example
+data source:
+
+```kotlin
+@Component
+class RandomNumberWidgetDataSource {
+
+    @WidgetDataSource(key = "randomNumber", title = "Random number")
+    fun randomNumberDataSource(properties: RandomNumberDataSourceProperties): RandomNumberResult {
+        return RandomNumberResult(Random.nextInt(properties.from, properties.to))
+    }
+}
+```
+
+The `@Component` is necessary to register the new data source as a Spring bean.
+
+The data source key `randomNumber` must be unique and is used the identify the data source. The same key must also be
+used in the front end data source.
+
+The function can be named anything.
+
+The function can have either 0 or 1 argument. The single argument must be an object that contains the same fields that
```

Comment:
I would add i little bit of explanation here. something like this:

If the datasource requires any configuration this will be passed in through the arguments of the function. All properties should be contained in a single object that is used as the argument for this function. The configuration from the front-end will be deserialized to this type, so the json structure should match that of the object. If no configuration is required the argument can be left out.